core/src/main/scala/kafka/metrics/KafkaMetricsGroup.scala

```diff
@@ -162,7 +162,7 @@ object KafkaMetricsGroup extends KafkaMetricsGroup with Logging {
   }
 
   private def toScope(tags: collection.Map[String, String]): Option[String] = {
-    val filteredTags = tags.filter { case (_, tagValue) => tagValue != ""}
+    val filteredTags = tags.filter { case (_, tagValue) => tagValue != null && !tagValue.isEmpty}
```

Comment:
@huxihx Thanks for the PR. I am concerned that this change could be masking a bug. Do we know what tag could be null so that we can fix that instead?

Reply:
@rajinisivaram It's kind of weird that the NPE should have not been thrown here since the given tags are all valid ApiKeys. I have no idea about why `tags` contains a null ApiKeys. This patch is more like a safe guard to protect against this which I think it's harmless somewhat. What do you think?

Reply:
@huxihx I had a look at the stack trace of the failure and I can't see how `tags` could contain a null value since they were used to create other metrics before the failing one. I think we do need to figure out the cause of the exception. So it would be better not to commit this change until we know the cause.